### PR TITLE
Fix savedsearch creation with AJAX search

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #10430

Fix creation of saved searches to be compatible with the new Fluid Search.

### CODE REVIEW

**Comment by cconard96** [comment]
Unrelated test causing failure. Another issue using current time in tests?

File: src/SavedSearch.php [hunk 1]
```diff
@@ -301,7 +301,7 @@ public function rawSearchOptions()
      */
     public function prepareSearchUrlForDB(array $input): array
     {
-        $taburl = parse_url(rawurldecode($input['url']));
+        $taburl = parse_url(html_entity_decode(rawurldecode($input['url'])));
```
Comment (cedric-anne) [inline src/SavedSearch.php:304]: Are you using `html_entity_decode` to unsanitize the input ? In this case, you should use:
```suggestion
        $taburl = parse_url(rawurldecode(Sanitizer::unsanitize($input['url'])));
```

**Comment by trasher** [comment]
> Unrelated test causing failure. Another issue using current time in tests?

Something like that, fixed as of #10472

**Comment by cconard96** [comment]
Hold off on merge/review. I may need to retest. See comment in linked issue.

**Comment by cconard96** [comment]
Seems OK

File: src/SavedSearch.php (hunk 1 above)
Comment (trasher) [reply src/SavedSearch.php:304]: ping @cconard96?

**Review by orthagh — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Comment by trasher** [comment]
You missed a `use use Glpi\Toolbox\Sanitizer;`

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**